pkg/controller/migmigration/postValidate.go

```diff
@@ -0,0 +1,298 @@
+
+	for _, gvk := range podManagingResources {
+		podManagersList.SetGroupVersionKind(gvk)
+		err := client.List(context.TODO(), options, &podManagersList)
+		if err != nil {
+			return ValidationFailed, err
+		}
+
+		for _, podManager := range podManagersList.Items {
+			expectedReplicas, found, err := unstructured.NestedFieldNoCopy(podManager.Object, "spec", "replicas")
+			if err != nil {
+				return ValidationFailed, err
+			}
+			if !found {
+				err = fmt.Errorf("Replicas field was not found in %s kind", gvk.String())
+				return ValidationFailed, err
+			}
+
+			readyReplicas, found, err := unstructured.NestedFieldNoCopy(podManager.Object, "status", "readyReplicas")
+			if err != nil {
+				return ValidationFailed, err
+			}
+			if found && expectedReplicas == readyReplicas {
+				continue
+			}
+
+			healthy, err := t.checkPodsHealth(client, options)
+			if err != nil || !healthy {
+				return ValidationFailed, err
+			}
+			return ValidationProgressing, nil
+		}
+	}
+	return ValidationSucceeded, nil
+}
+
+func (t *Task) validateJobs(client k8sclient.Client, options *k8sclient.ListOptions) (bool, error) {
+	jobList := batchv1.JobList{}
+
+	err := client.List(context.TODO(), options, &jobList)
+	if err != nil {
+		return false, err
+	}
+
+	for _, job := range jobList.Items {
+		if job.Status.Failed == 0 {
+			continue
+		}
+
+		// Check the state of pods
+		healthy, err := t.checkPodsHealth(client, options)
+		if err != nil {
+			return false, err
+		}
+		if !healthy {
+			return false, err
+		}
+		break
+	}
+	return true, nil
+}
+
+func (t *Task) checkPodsHealth(client k8sclient.Client, options *k8sclient.ListOptions) (bool, error) {
+	failedPodCounter := 0
+	pendingPodCounter := 0
+	podNames := []string{}
+	podList := corev1.PodList{}
+	eventList := corev1.EventList{}
+
+	err := client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return false, err
+	}
+
+	err = client.List(context.TODO(), options, &eventList)
+	if err != nil {
+		return false, err
+	}
+
+	for _, pod := range podList.Items {
+		switch pod.Status.Phase {
+		case corev1.PodSucceeded, corev1.PodRunning:
+			continue
+		case corev1.PodFailed:
+			failedPodCounter++
+		default:
+			pendingPodCounter++
+		}
+
+		podNames = append(podNames, pod.GetName())
+	}
+
+	correlatedEvents := collectRelatedEvents(podNames, eventList.Items)
+	eventsJSON, err := json.MarshalIndent(correlatedEvents, "", " ")
+	if err != nil {
+		return false, err
+	}
+
+	if failedPodCounter > 0 {
+		t.Owner.Status.SetCondition(migapi.Condition{
+			Type:     Validating,
+			Status:   True,
+			Reason:   t.Phase,
+			Category: Critical,
+			Message:  string(eventsJSON),
+		})
+	} else {
+		t.Owner.Status.SetCondition(migapi.Condition{
+			Type:     Validating,
+			Status:   True,
+			Reason:   t.Phase,
+			Category: Advisory,
+			Message:  string(eventsJSON),
+		})
+	}
+
+	t.Owner.Status.SetCondition(migapi.Condition{
+		Type:     ValidationDetails,
```

Comment:
A separate `details` condition is a bit odd.  Let's consider a single condition.  Our conditions have an `Items []string` field that is handy for this.

Thinking something like this:

```
t.Owner.Status.SetCondition(migapi.Condition{
    Type:     Verification,
    Status:   True,
    Reason:   Running,
    Category: Advisory,
    Message:  "Verification running."
    Items: []string{}
})
```
Then when a Pod fails verification.
```
condition := t.Owner.Status.FindCondition(Verification)
for _, event := range events {
    condition.Items = append(condition.Items, string(event-json))
}
```

At the end,
```
condition := t.Owner.Status.FindCondition(Verification)
if len(condition.Items) > 0 {
    condition.Category = Critical
    condition.Reason = Failed
    condition.Message = "Verification failed: []"  // The [] is replaced Items by when staging ends.
} else {
    condition.Reason = Succeeded
    condition.Message = "Verification succeeded." 
}
```

Reply:
As we discussed: Rather than listing events, let's provide the user with a list of resources that failed (Deployment|ReplicaSet, ..)
```
condition := t.Owner.Status.FindCondition(Verification)
for _, r := range resources {
    ref := fmt.Sprintf("%s/%s:%s", r.Kind, r.Namespace, r.Name)
    condition.Items = append(condition.Items, ref)
}
```